test/position-managers/PositionManager.t.sol

```diff
@@ -0,0 +1,263 @@
+            amount1Desired
+        );
+
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: tickLower, tickUpper: tickUpper});
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        uint256 tokenId = lpm.nextTokenId();
+        BalanceDelta delta = mint(range, liquidityToAdd, address(this), ZERO_BYTES);
+
+        uint256 balance0After = currency0.balanceOfSelf();
+        uint256 balance1After = currency1.balanceOfSelf();
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(1), address(this));
+
+        assertEq(uint256(int256(-delta.amount0())), amount0Desired);
+        assertEq(uint256(int256(-delta.amount1())), amount1Desired);
+        assertEq(balance0Before - balance0After, uint256(int256(-delta.amount0())));
+        assertEq(balance1Before - balance1After, uint256(int256(-delta.amount1())));
+    }
+
+    function test_mint_recipient(IPoolManager.ModifyLiquidityParams memory seedParams) public {
+        IPoolManager.ModifyLiquidityParams memory params = createFuzzyLiquidityParams(key, seedParams, SQRT_PRICE_1_1);
+        uint256 liquidityToAdd =
+            params.liquidityDelta < 0 ? uint256(-params.liquidityDelta) : uint256(params.liquidityDelta);
+
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        uint256 tokenId = lpm.nextTokenId();
+        mint(range, liquidityToAdd, address(alice), ZERO_BYTES);
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(tokenId), alice);
+    }
+
+    function test_burn(IPoolManager.ModifyLiquidityParams memory params) public {
+        uint256 balance0Start = currency0.balanceOfSelf();
+        uint256 balance1Start = currency1.balanceOfSelf();
+
+        // create liquidity we can burn
+        uint256 tokenId;
+        (tokenId, params) = addFuzzyLiquidity(lpm, address(this), key, params, SQRT_PRICE_1_1, ZERO_BYTES);
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(1), address(this));
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta));
+
+        // burn liquidity
+        uint256 balance0BeforeBurn = currency0.balanceOfSelf();
+        uint256 balance1BeforeBurn = currency1.balanceOfSelf();
+
+        BalanceDelta deltaDecrease = decreaseLiquidity(tokenId, liquidity, ZERO_BYTES);
+        burn(tokenId);
+
+        (liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, 0);
+
+        // TODO: slightly off by 1 bip (0.0001%)
```

Comment:
weirdddd

Reply:
this is old.. i changed it to exactly equal and its fixed. the comment below though is true and I believe its a known rounding in pulling out liquidity,